addons/io/org.openhab.io.azureiothub/src/main/java/org/openhab/io/internal/azureiothub/CloudClient.java

```diff
@@ -0,0 +1,169 @@
+        this.commandEnabled = commandEnabled;
+        this.eventPublisher = eventPublisher;
+
+        azureDeviceStore = new AzureDevices(this.connectionstring);
+
+        if (this.commandEnabled) {
+            azureClient = receiveMessages("0");
+        }
+    }
+
+    public void sendItemUpdate(String deviceId, String state) {
+
+        AzureDevice device;
+        try {
+            device = azureDeviceStore.getDevice(deviceId);
+        } catch (Exception e) {
+            logger.error("Failed to obtain azure device", e);
+            return;
+        }
+
+        setItemState(device, state);
+    }
+
+    private void setItemState(AzureDevice azureDevice, String state) {
+        AzureEventCallback callback = new AzureEventCallback();
+
+        AzureDatapoint datapoint = new AzureDatapoint();
+
+        Device device = azureDevice.getDevice();
+        if (device == null) {
+            logger.error("Invalid device connection for device, can not send item state update");
+            return;
+        }
+
+        datapoint.deviceId = device.getDeviceId();
+        datapoint.value = state;
+
+        Message msg = new Message(datapoint.serialize());
+
+        azureDevice.sendMessage(msg, callback, lockobj);
+
+        synchronized (lockobj) {
+            try {
+                lockobj.wait();
+            } catch (InterruptedException e) {
+                // done
+            }
+        }
+    }
+
+    public void shutdown() {
+        if (azureClient != null) {
+            try {
+                azureClient.closeSync();
+            } catch (ServiceBusException e) {
+                logger.error("failed to close IoT client", e);
+            }
+        }
+    }
+
+    private EventHubClient receiveMessages(final String partitionId) {
+        EventHubClient client = null;
+        try {
+            client = EventHubClient.createFromConnectionStringSync(this.connectionstring);
+        } catch (Exception e) {
+            logger.error("Failed to create IoT EventHub client", e);
+            return null;
+        }
+        try {
+            client.createReceiver(EventHubClient.DEFAULT_CONSUMER_GROUP_NAME, partitionId, Instant.now())
+                    .thenAccept(new Consumer<PartitionReceiver>() {
+                        @Override
+                        public void accept(PartitionReceiver receiver) {
+                            logger.debug("Created IOT Hub receiver on partition {}", partitionId);
+                            try {
+                                while (true) {
+                                    Iterable<EventData> receivedEvents = receiver.receive(100).get();
+                                    int batchSize = 0;
+                                    if (receivedEvents != null) {
+                                        for (EventData receivedEvent : receivedEvents) {
+                                            logger.debug("Offset: {}, SeqNo: {}, EnqueueTime: {}",
```

Comment:
Please consider splitting this method up, the nesting is too deep. The minimum you should do is extract the body to a method with a meaningful name.